pkgs/stdenv/native/default.nix

```diff
@@ -1,4 +1,9 @@
-{ system, allPackages ? import ../../.., config }:
+{ _defaults    ? import ../debug.nix
+, argsResolved ? _defaults.extend (_: _: args)
+, allPackages  ? argsResolved.allPackages
```

Comment:
I agree in principle, but here are two headaches:

1. One of the nixpkgs tests is building darwin bootstrap tools and booting a stdenv with it, so I need a decent portion of this complexity just to make an `allPackages` argument suitable for that test to pass.

2. I beleive the goal wrt debugging is less tracing than being able to load a small subset up, potentially with different arguments. That's also why these files are big `rec { .. }` rather than `let .. in ..`.

@copumpkin I think you are the person to ask via both of these, and more broadly what to do with this final commit.